algorithms/gr-dfsrh/java/DepthFirstSearch.java

```diff
@@ -0,0 +1,73 @@
+import java.io.*; 
+import java.util.*; 
+  
+// This class represents a directed graph using adjacency list 
+// representation 
+class Graph 
+{ 
+    private int V;   // No. of vertices 
+  
+    // Array  of lists for Adjacency List Representation 
+    private LinkedList<Integer> adj[]; 
+  
+    // Constructor 
+    @SuppressWarnings("unchecked") 
+    Graph(int v) 
+    { 
+        V = v; 
+        adj = new LinkedList[v]; 
+        for (int i=0; i<v; ++i) 
+            adj[i] = new LinkedList(); 
+    } 
+  
+    //Function to add an edge into the graph 
+    void addEdge(int v, int w) 
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6327382)